### PR TITLE
String manipulation

### PR DESCRIPTION
**Pull request by abuzaforfagun**
- String reverse
- Common prefix
- Common suffix

This will resolve the issue #12 

## Summary by CodeRabbit

- **New Features**
	- Introduced new string manipulation functions: 
		- `Reverse` to reverse strings while handling Unicode characters.
		- `CommonPrefix` to find the longest common prefix among multiple strings.
		- `CommonSuffix` to identify the longest common suffix among multiple strings.

- **Tests**
	- Added unit tests for the `Reverse`, `CommonPrefix`, and `CommonSuffix` functions to ensure accuracy with various input cases, including accented characters and empty strings.

### CODE REVIEW

File: strings/strings.go [hunk 1]
```diff
@@ -156,3 +156,85 @@ func SanitizeEmail(email string) string {
 
 	return strings.TrimSpace(email)
 }
+
+func Reverse(input string) string {
```
Comment (kashifkhan0771) [inline strings/strings.go:160]: I would recommend to add documentation comments on top of function as also suggested by codeRabbit.
**Note:** Not all comments from codeRabbit need to be resolved unless it's something critical or is something which is standard in library.

File: strings/strings_test.go [hunk 2]
```diff
@@ -322,3 +322,113 @@ func TestTitle(t *testing.T) {
 		})
 	}
 }
+
+func TestReverse(t *testing.T) {
```
Comment (kashifkhan0771) [inline strings/strings_test.go:326]: Test cases are very detailed ❤️

**Comment by abuzaforfagun** [comment]
@coderabbitai review

**Review by kashifkhan0771 — APPROVED** [review]
LGTM! Thanks @abuzaforfagun for the contribution.

**Pull request merged**